sql/tables.sql

```diff
@@ -59,7 +59,7 @@ CREATE TABLE IF NOT EXISTS _timescaledb_catalog.tablespace (
 SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.tablespace', '');
 
 -- A dimension represents an axis along which data is partitioned.
-CREATE TABLE _timescaledb_catalog.dimension (
+CREATE TABLE IF NOT EXISTS _timescaledb_catalog.dimension (
```

Comment:
First off, this makes things consistent since some tables/objects already had `IF NOT EXISTS`. Second, why shouldn't it be part of upgrade scripts? Can you clarify the reason why not and what makes this file special? Third, this is in fact needed so that one can change the catalog schema in more advanced ways than simply removing/adding columns without having to mirror a full `CREATE TABLE` in the upgrade script, which might be more error prone.

Reply:
Just to be clear, in the upgrade script I rename the old chunk_index table, then let the tables.sql install the new/updated table, then in the post script I convert data from the old table to the new table, finally dropping the old table. This avoids having to specify an identical full CREATE TABLE statement in the upgrade script. It felt messy (if even possible) to ALTER TABLE my way through an update to the table that removes columns and converts  data simultaneously. Maybe it is possible, but it also leaves the table with a lot of "garbage" column definitions. I also think that risks messing up the row index mapping in catalog.c, because of the garbage columns. The only safe ALTER TABLE is to append columns at the end.

Including this file seems similar to what happens with functions on other objects during upgrades (i.e., we include the full source code, irrespective of whether there have been modifications or not).

Reply:
I recall us having lengthy discussions about this and IIRC the conclusion was that we should avoid `IF NOT EXISTS`. I believe one reason was that it will not do any checks on the actual format of the table, only that there exists a table of that name.

Reply:
I don't think I was involved in this discussion so I don't know the context. Still, I don't see why the lack of validation when using `IF NOT EXISTS` would be a problem as it would be no worse than before and this is not a mechanism to replace tables anyway (i.e., to replace a table you need first to remove it in an update script). Without including this file in the update, we'd have to specify two identical  `CREATE TABLE` statements in two places. I can certainly do this but I still haven't heard a really good reason for why we are doing things this way. Right now the update script compilation is quite confusing, SQL files are listed across three different `.txt` files and there are no docs describing their function apart from vague naming.